src/ProjectsProvider.js

```diff
@@ -26,9 +31,69 @@ export const useProjects = () => {
   return context;
 };
 
+const prepareStaticProjects = (projects) => {
+  return projects.map((el) => {
+    return {
+      id: uuidv4(),
+      name: el.name,
+      description: el.description,
+      imageUrl: el.imageUrl,
+      link: el.link ? BASE_URL + el.link : el.link,
+      tags: el.tags,
+      topic: el.topic,
+      quarter: el.quarter,
+      coordinates: [
+        Number(el.gps.split(',')[0]),
+        Number(el.gps.split(',')[1].trim()),
+      ],
+      date: format(
+        parse('', '', new Date(el.timestamp.replace(' ', 'T'))),
+        'dd.mm.yyyy'
+      ),
+      type: el.type,
+    };
+  });
+};
+
+
 export const ProjectsProvider = ({ children }) => {
   let [searchParams] = useSearchParams();
 
+  const [mapTopicIdToTopic, setMapTopicIdToTopic] = React.useState(null);
+  React.useEffect(() =>
+    fetch(`${BASE_URL}/wp-json/wp/v2/project-category/`)
+      .then(res => res.json())
+      .then(data => setMapTopicIdToTopic(new Map(data.map(obj => [obj.id, obj.name]))))
+    , []);
+
+  const [wordpressProjectsRaw, setWordpressProjectsRaw] = React.useState([]);
+  React.useEffect(() => fetch(`${BASE_URL}/wp-json/wp/v2/projects`)
+    .then(res => res.json())
+    .then(data => setWordpressProjectsRaw(data.filter(el => el.acf && el.acf.map))),
+    []);
+
+  const [wordpressProjects, setWordpressProjects] = React.useState([]);
+  React.useEffect(() =>
+    setWordpressProjects(wordpressProjectsRaw.map(el => Object({
```

Comment:
issue: 
you probably need this 
React.useEffect(async() =>
    setWordpressProjects(
await Promise.all(wordpressProjectsRaw.map(async el =>

check 
https://stackoverflow.com/questions/33438158/best-way-to-call-an-asynchronous-function-within-map

also first prepare/normalise data and after that set it in the state

Reply:
Used Promise.all as suggested and explained. I think we are good to do it this way as technically it happens only once.